### PR TITLE
fix: suggestion list CSS changes

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
# fix: suggestion list CSS changes

Some changes suggested by design on review

### CODE REVIEW

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue [hunk 1]
```diff
@@ -89,10 +91,26 @@ export default {
...
     upHandler () {
       this.selectedIndex = ((this.selectedIndex + this.items.length) - 1) % this.items.length;
+
+      const activeElement = this.$refs.suggestionList.querySelector('.dt-list-item--highlighted');
+      if (activeElement) {
+        activeElement.scrollIntoView({
+          behaviour: 'smooth',
+          block: 'center',
+        });
+      }
```
Comment (braddialpad) [inline components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: Instead of repeating in both functions, isolate in a single function `scrollActiveElementIntoView` or somethin like that.

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue [hunk 2]
```diff
@@ -24,6 +25,7 @@
   </div>
 </template>
 
+import type { resolveFocusPosition } from '@tiptap/vue-3';
```
Comment (braddialpad) [inline components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: does this need to be used?

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue [hunk 3]
```diff
@@ -2,14 +2,15 @@
   <div class="d-popover__dialog">
     <ul
       v-show="items.length"
+      ref="suggestionList"
       class="dt-suggestion-list"
     >
       <dt-list-item
         v-for="(item, index) in items"
         :key="index"
```
Comment (braddialpad) [inline components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: Don't use index for key here it's actually causing a warning to be thrown in vue. Need to use some string that uniquely identifies the item.

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue
Comment (braddialpad) [inline components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: Hmm seems like the suggestion list is not working anymore when you type @? Actually doesn't seem related to this PR at all it's happening on vue.dialpad.design/vue3 as well. Pretty strange error...

Screenshot 2023-10-16 at 10 10 29 AM

**Review by braddialpad — CHANGES_REQUESTED** [review]

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue (hunk 2 above)
Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: ah my bad, didnt notice this sneak in , vs code autocomplete :(

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue
Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: the @ works only on the third storybook with channel mentions and its working. Let me see recent changes to check if this is introduced recently. i cant find anything in stacktrace

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: damn its because of this comment on the docstring:
```
The only required key is the items function which is used to query the contacts for suggestion.
     * items({ query }) => { return [<ContactObj>]; }
     * ContactObj format:
     * { name: string, avatarSrc: string, contactId: string }
```

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: fixed it

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: Not working once it's deployed. Getting this:

```
[Vue warn]: Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead.
```

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: i think its another unrelated issue, taking a look

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue (hunk 3 above)
Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: Fix was not exactly what I meant. Use contactId for the key instead of index.

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue
Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: Does the @ list show up for you though?

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: Yes in the storybook called with mention suggestion. Type @T and it will show up.

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: ohh got ya. Didn't know I needed to type "t"

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: We should put a string above the component in the example saying how to do this, because nobody is going to know they have to put "t" if we don't tell them.

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: LEt me change the filter function to return all the list when its empty. That would be better

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue (hunk 3 above)
Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: the problem is that this component is used for contact suggestion, emoji suggestion and in future channel suggestion. So probab;y have to pass a unique index ?

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue
Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue:1]: Yeah thats fine

File: components/rich_text_editor/extensions/suggestion/SuggestionList.vue (hunk 3 above)
Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: btw the warning you talking about is this:
```
[Vue warn]: Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead.
```

this is because of listening to events on blur/focus/update and this has been happening since before i took this component. 
```
/**
     * The Editor exposes event hooks that we have to map our emits into. See
     * https://tiptap.dev/api/events for all events.
     */
    addEditorListeners () {
      // The content has changed.
      this.editor.on('update', () => {
        const value = this.getOutput();
        this.$emit('input', value);
        this.$emit('update:modelValue', value);
      });

      // The editor is focused.
      this.editor.on('focus', ({ event }) => {
        this.$emit('focus', event);
      });

      // The editor isn’t focused anymore.
      this.editor.on('blur', ({ event }) => {
        this.$emit('blur', event);
      });
    },
```

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: Then make it a generic id field that all of these would use

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: The problem with using index, is it changes if the order of the item in the list changes, then things get all kinds of messed up with reactivity and such, so it's best practice to not use it.

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: > this is because of listening to events on blur/focus/update and this has been happening since before i took this component.

All good, not relevant to this PR then.

Comment (sanjeevdialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: So i used item.id and i changed the emoji implementation to add a atrribute of id. Unfortunately we dont have a unique id for emoji, so i made id = unicode character which will be unique in either way. Now key can just refer item.id.

Comment (braddialpad) [reply components/rich_text_editor/extensions/suggestion/SuggestionList.vue]: yeah thats fine, id just has to be SOMETHING unique even if it's a unicode char

**Review by braddialpad — APPROVED** [review]
